Failed to compile using GCC 11.2.1

The document states that only GCC >= 9 is required.

https://github.com/zeek/spicy/blob/ab69192894caef32e7bca316ec6fa5e65d4a40a4/doc/installation.rst#L210

Error log:
```bash
/spicy/spicy/runtime/src/base64.cc: In member function ‘hilti::rt::Bytes spicy::rt::base64::Stream::encode(const hilti::rt::Bytes&)’:
/spicy/spicy/runtime/src/base64.cc:38:40: error: call of overloaded ‘basic_string(SafeInt<long unsigned int, hilti::rt::integer::detail::SafeIntException>, <brace-enclosed initializer list>)’ is ambiguous
   38 |     std::string buf(data.size() * 2, {});
      |                                        ^
In file included from /opt/rh/devtoolset-11/root/usr/include/c++/11/string:55,
                 from /opt/rh/devtoolset-11/root/usr/include/c++/11/stdexcept:39,
                 from /spicy/hilti/runtime/include/hilti/rt/types/bytes.h:7,
                 from /spicy/spicy/runtime/src/base64.cc:3:
/opt/rh/devtoolset-11/root/usr/include/c++/11/bits/basic_string.h:3734:9: note: candidate: ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&) [with _InputIterator = SafeInt<long unsigned int, hilti::rt::integer::detail::SafeIntException>; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
 3734 |         basic_string(_InputIterator __beg, _InputIterator __end,
      |         ^~~~~~~~~~~~
/opt/rh/devtoolset-11/root/usr/include/c++/11/bits/basic_string.h:3673:7: note: candidate: ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::basic_string<_CharT, _Traits, _Alloc>::size_type, _CharT, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
 3673 |       basic_string(size_type __n, _CharT __c, const _Alloc& __a = _Alloc())
      |       ^~~~~~~~~~~~
/spicy/spicy/runtime/src/base64.cc: In member function ‘hilti::rt::Bytes spicy::rt::base64::Stream::decode(const hilti::rt::Bytes&)’:
/spicy/spicy/runtime/src/base64.cc:63:40: error: call of overloaded ‘basic_string(SafeInt<long unsigned int, hilti::rt::integer::detail::SafeIntException>, <brace-enclosed initializer list>)’ is ambiguous
   63 |     std::string buf(data.size() * 2, {});
      |                                        ^
In file included from /opt/rh/devtoolset-11/root/usr/include/c++/11/string:55,
                 from /opt/rh/devtoolset-11/root/usr/include/c++/11/stdexcept:39,
                 from /spicy/hilti/runtime/include/hilti/rt/types/bytes.h:7,
                 from /spicy/spicy/runtime/src/base64.cc:3:
/opt/rh/devtoolset-11/root/usr/include/c++/11/bits/basic_string.h:3734:9: note: candidate: ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&) [with _InputIterator = SafeInt<long unsigned int, hilti::rt::integer::detail::SafeIntException>; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
 3734 |         basic_string(_InputIterator __beg, _InputIterator __end,
      |         ^~~~~~~~~~~~
/opt/rh/devtoolset-11/root/usr/include/c++/11/bits/basic_string.h:3673:7: note: candidate: ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::basic_string<_CharT, _Traits, _Alloc>::size_type, _CharT, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
 3673 |       basic_string(size_type __n, _CharT __c, const _Alloc& __a = _Alloc())
      |       ^~~~~~~~~~~~
```

--- a/spicy/runtime/src/base64.cc
+++ b/spicy/runtime/src/base64.cc
@@ -35,8 +35,8 @@ hilti::rt::Bytes Stream::encode(const hilti::rt::Bytes& data) {
     if ( ! _state )
         throw Base64Error("encoding already finished");
 
-    std::string buf(data.size() * 2, {});
-    auto len = base64_encode_block(data.data(), static_cast<int>(data.size()), buf.data(), &_state->estate);
+    std::string buf(static_cast<std::string::size_type>(data.size() * 2), {});
+    auto len = base64_encode_block(data.data(), static_cast<std::size_t>(data.size()), buf.data(), &_state->estate);
     return hilti::rt::Bytes(buf.data(), len);
 }
 
@@ -60,8 +60,8 @@ hilti::rt::Bytes Stream::decode(const hilti::rt::Bytes& data) {
     if ( ! _state )
         throw Base64Error("decoding already finished");
 
-    std::string buf(data.size() * 2, {});
-    auto len = base64_decode_block(reinterpret_cast<const char*>(data.data()), static_cast<int>(data.size()),
+    std::string buf(static_cast<std::string::size_type>(data.size() * 2), {});
+    auto len = base64_decode_block(reinterpret_cast<const char*>(data.data()), static_cast<std::size_t>(data.size()),
                                    buf.data(), &_state->dstate);
 
     return hilti::rt::Bytes(buf.data(), len);
@@ -75,8 +75,8 @@ hilti::rt::Bytes Stream::decode(const hilti::rt::stream::View& data) {
 
     for ( auto block = data.firstBlock(); block; block = data.nextBlock(block) ) {
         std::string buf(block->size * 2, {});
-        auto len = base64_decode_block(reinterpret_cast<const char*>(block->start), static_cast<int>(block->size),
-                                       buf.data(), &_state->dstate);
+        auto len = base64_decode_block(reinterpret_cast<const char*>(block->start),
+                                       static_cast<std::size_t>(block->size), buf.data(), &_state->dstate);
         decoded.append(hilti::rt::Bytes(buf.data(), len));
     }
 

